agent/runner/actions/mongodb_explain_action.go

```diff
@@ -31,23 +31,32 @@ import (
 	"github.com/percona/pmm/api/agentpb"
 )
 
+const mongoDBExplainActionType = "mongodb-explain"
+
 type mongodbExplainAction struct {
 	id      string
 	timeout time.Duration
 	params  *agentpb.StartActionRequest_MongoDBExplainParams
 	tempDir string
+	dsn     string
 }
 
 var errCannotExplain = fmt.Errorf("cannot explain this type of query")
 
 // NewMongoDBExplainAction creates a MongoDB EXPLAIN query Action.
-func NewMongoDBExplainAction(id string, timeout time.Duration, params *agentpb.StartActionRequest_MongoDBExplainParams, tempDir string) Action {
+func NewMongoDBExplainAction(id string, timeout time.Duration, params *agentpb.StartActionRequest_MongoDBExplainParams, tempDir string) (Action, error) {
+	dsn, err := templates.RenderDSN(params.Dsn, params.TextFiles, filepath.Join(tempDir, strings.ToLower(mongoDBExplainActionType), id))
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
```

Comment:
I'm afraid that in case of SSL enabled it will be difference for each action. I would recommend to use params.Dsn instead of rendered one

Reply:
Mmm, I didn't get it. This chunk was extracted from `Run` method just to be able to access `dsn` later (in ` DSN()` method).

Reply:
We discussed this topic on call.